### ISSUE TITLE
Increase Assets API performance

### ISSUE THREAD

**Issue by bartfeenstra** [bartfeenstra/betty#1518]
- Benchmark everything!
- Consider renaming the file system API to the assets API
- Consider removing functionality not needed for assets management (what exactly?)
- Consider scanning for all files upon first use and then caching the resolved results, e.g. removing overridden files from the resulting file list.
- FileSystem.open() should only take a single path for consistency (also Betty never uses the method itself)
- Make AssetRepository an interface and refactor the existing class into a concrete implementation.
- add a decorator for Project.assets

## Blocked by
- https://github.com/bartfeenstra/betty/pull/1613
